src/andromede/simulation/optimization.py

```diff
@@ -579,161 +678,192 @@ def make_constraint(
-
-                    # TODO: Add BoolVar or IntVar if the variable is specified to be integer or bool
-                    solver_var = solver.NumVar(lower_bound, upper_bound, model_var.name)
-                    component_context.add_variable(block_timestep, scenario, solver_var)
-
-
-def _register_connection_fields_definitions(
-    network: Network, opt_context: OptimizationContext
-) -> None:
-    for cnx in network.connections:
-        for field_name in list(cnx.master_port.keys()):
-            master_port = cnx.master_port[field_name]
-            port_definition = master_port.component.model.port_fields_definitions.get(
-                PortFieldId(port_name=master_port.port_id, field_name=field_name.name)
-            )
-            expression_node = port_definition.definition  # type: ignore
-            instantiated_expression = add_component_context(
-                master_port.component.id, expression_node
-            )
-            opt_context.register_connection_fields_expressions(
-                component_id=cnx.port1.component.id,
-                port_name=cnx.port1.port_id,
-                field_name=field_name.name,
-                expression=instantiated_expression,
-            )
-            opt_context.register_connection_fields_expressions(
-                component_id=cnx.port2.component.id,
-                port_name=cnx.port2.port_id,
-                field_name=field_name.name,
-                expression=instantiated_expression,
-            )
 
+    def _create_variables(self) -> None:
+        for component in self.context.network.all_components:
+            component_context = self.context.get_component_context(component)
+            model = component.model
 
-def _create_constraints(
-    network: Network, opt_context: OptimizationContext, solver: lp.Solver
-) -> None:
-    for component in network.all_components:
-        for constraint in component.model.get_all_constraints():
-            instantiated_expr = _instantiate_model_expression(
-                constraint.expression, component.id, opt_context
-            )
-            instantiated_lb = _instantiate_model_expression(
-                constraint.lower_bound, component.id, opt_context
-            )
-            instantiated_ub = _instantiate_model_expression(
-                constraint.upper_bound, component.id, opt_context
-            )
-            instantiated_constraint = Constraint(
-                name=constraint.name,
-                expression=instantiated_expr,
-                lower_bound=instantiated_lb,
-                upper_bound=instantiated_ub,
-            )
-            _create_constraint(
-                solver,
-                opt_context.get_component_context(component),
-                instantiated_constraint,
-            )
+            for model_var in model.variables.values():
+                if not _should_keep_in_model(self.problem_type, model_var.context):
```

Comment:
Much clearer but I still find it even clearer to do `if _should_keep_in_model():` even if it makes one more indentation

Reply:
🙃

Reply:
Done!